gpAux/gpperfmon/src/gpmon/gpmon_agg.c

```diff
@@ -1504,7 +1504,15 @@ static apr_uint32_t write_qlog_full(FILE* fp, qdnode_t *qdnode, const char* nows
 	    fprintf(fp, "|");
         bytes_written++;
 
-        if (!all_good || iter == 1){
+        if (iter == 0)
+        {
+            fputc('0', fp);
```

Comment:
This code is executed always one time. So I suggest to put it before the loop and do not change the loop.
```
fputs("|0", fp);
bytes_written+=2;
```